scripts/azdo/update_sstate_cache.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+usage() {
+	local exit_code=${1:-2}
+	test $exit_code -eq 0 || exec 1>&2
+
+	cat <<EOF
+
+Syntax:
+$(basename $BASH_SOURCE) push local_sstate_cache_dir argo_sstate_cache_dir
+   Push the local sstate-cache directory contents to the remote argo
+   server directory.
+
+$(basename $BASH_SOURCE) fetch local_sstate_cache_dir argo_sstate_cache_dir
+   Fetch the latest sstate cache from the remote argo cache directory
+   and install to the local sstate-cache directory.
+
+$(basename $BASH_SOURCE) -h
+   Print this help message and exit
+
+Arguments:
+local_sstate_cache_dir      Path to the local sstate cache directory
+argo_sstate_cache_dir       Path to the remote argo sstate cache directory
+
+EOF
+	exit $exit_code
+}
+
+push_cache() {
+	if [ ! -d $local_sstate_cache_dir -o -z "$(ls -A $local_sstate_cache_dir 2>/dev/null)" ]; then
+		echo $local_sstate_cache_dir is missing/empty. Please provide a valid local_sstate_cache_dir name with sstate-cache contents.
+		usage
+	fi
+
+	mkdir -p $argo_sstate_cache_dir
+	pushd $argo_sstate_cache_dir >/dev/null
+
+	# Copy the contents of the cache to a temp .latest directory
+	# before removing the .latest suffix from the name.
+	temp_argo_dir=$(date +"%s").latest
+	mkdir $temp_argo_dir
+	echo Pushing sstate cache to argo at $temp_argo_dir
+	cp -rL $local_sstate_cache_dir/* $temp_argo_dir/
+	mv $temp_argo_dir ${temp_argo_dir%.latest}
```

Comment:
Note that because this is on argo, the `mv` operation may not be atomic, and there is a possibility an incomplete cache could be consumed by build jobs. I could instead:

1. Create a temp file in the local cache directory before starting the `cp` operation, say `donotuse`
2. Remove the file in the remote directory after the `mv` operation has completed 
3. Update the `fetch_cache` operation to ignore any directory with the .latest suffix, as well a `donotuse` file

Lmk what you think.

Reply:
I think the current script does what you described, albeit using a .latest suffix and not .tmp? Is there a different mechanism to remove the suffix if not with the `mv` command?
With `mv`, I was thinking that if the sequence turned out such that a new directory (with the suffix removed) gets created before the contents of the old directory are moved to the new directory, there is a chance that when fetch is called it picks up the new directory because it doesn't have the suffix, but the cache contents in the new directory would not be complete for a while.

Reply:
@amstewart what do you think?

Reply:
I think I got a little lost in the discussion last week. I have a more-clear picture of what you're talking about now and I think the `donotuse` poison-pill idea is a pretty elegant solution. ACK from me. :+1: